core/templates/pages/exploration-editor-page/modal-templates/state-diff-modal.component.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for state diff modal.
+ */
+
+import { OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { State } from 'domain/state/StateObjectFactory';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ContextService } from 'services/context.service';
+import { StateDiffModalBackendApiService } from '../services/state-diff-modal-backend-api.service';
+
+@Component({
+  selector: 'state-diff-modal',
+  templateUrl: './state-diff-modal.component.html',
+})
+export class StateDiffModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+    newState: State | null;
+    oldState: State | null;
+    newStateName: string;
+    oldStateName: string;
+    headers: {
+        leftPane: string;
+        rightPane: string;
+    };
+    yamlStrs: {
+        leftPane: string;
+        rightPane: string;
+    } = {
+      leftPane: '',
+      rightPane: '',
+    };
+    CODEMIRROR_MERGEVIEW_OPTIONS = {
+      lineNumbers: true,
+      readOnly: true,
+      mode: 'yaml',
+      viewportMargin: 100
+    };
```

Comment:
Can you add a interface for this variable.

Reply:
done